src_py/sysfont.py

```diff
@@ -458,7 +453,9 @@ def get_fonts():
     """
     if not Sysfonts:
         initsysfonts()
-    return list(Sysfonts)
+
+    # 'Sysfonts' can contain a sentinel 'None' key, don't forward that to users
+    return [i for i in Sysfonts if i is not None]
```

Comment:
I haven't read through this at all, but isn't this just from one backend? Shouldn't the filtering happen at the level of that backend then?

Reply:
hmmm no. `Sysfonts` is the global state used by all backends. And I think the approach of checking here before returning is safer that putting many more checks elsewhere

Reply:
Sorry, unclear. I mean, if the sysfonts unix backend is the only one putting `None`s in, it should be sanitized there. Other bits of code besides `get_fonts()` fetch stuff from the global dictionary.

Reply:
`initsysfonts` calls system specific inits and uses `None` sentinel on all platforms...

Reply:
I see. 
Sysfont is either a valid dict of fonts or a dict that exclusively contains the key value pair None = None.

I'm not sure this causes any problems, but it seems a little strange. It seems cleaner to me to not try to contain this extra "is init" state in the Sysfont dict, but rather have an init flag that gets set. What do you think?

Reply:
Ye I did think of that too, but I scrapped that idea to make this PR a bit easier to review 😉 
Now that you mention it... maybe it's not going to create too much diff so it might be worth it if it provides that extra clarity

Reply:
Well... I'm inclined to postpone this to a future refactor of sysfont code, I'm not going to try to refactor things in this PR since this is primarily a bugfix PR

Reply:
I disagree. Dealing with `None`s is the whole point of this PR, it's not an incidental refactoring.

Reply:
aight, I made this change